docs/assets/scripts/example-knn-image.js

```diff
@@ -4,93 +4,67 @@ KNN Image Classifier Demo
 ===
 */
 
-let knn;
-let video;
+// Set up the video stream
+const video = document.getElementById('video');
+navigator.getUserMedia({ video: true }, handleVideo, videoError);
 
-function preload() {
-  // Initialize the KNN method.
-  knn = new ml5.KNNImageClassifier(modelLoaded, 2, 1);
+function handleVideo(stream) {
+  video.src = window.URL.createObjectURL(stream);
 }
 
-function setup() {
-  createCanvas(320, 240).parent('canvasContainer');
-  video = createCapture(VIDEO);
-  background(0);
-  video.size(227, 227);
-  video.hide();
-
-  // Train buttons
-  buttonA = select('#buttonA');
-  buttonA.mousePressed(() => {
-    train(1);
-  });
-
-  buttonB = select('#buttonB');
-  buttonB.mousePressed(() => {
-    train(2);
-  });
-
-  // Reset buttons
-  resetBtnA = select('#resetA');
-  resetBtnA.mousePressed(() => {
-    clearClass(1);
-    updateExampleCounts();
-  });
-
-  resetBtnB = select('#resetB');
-  resetBtnB.mousePressed(() => {
-    clearClass(2);
-    updateExampleCounts();
-  });
-
-  buttonPredict = select('#buttonPredict');
-  buttonPredict.mousePressed(predict);
-}
-
-function draw() {
-  background(0);
-  image(video, 0, 0, width, height);
+function videoError() {
+  console.error('Video not available');
 }
 
 // A function to be called when the model has been loaded
 function modelLoaded() {
-  select('#loading').html('Model loaded!');
+  document.getElementById('loader').innerText = 'Model loaded!';
```

Comment:
Since we are using p5 for video capture we might as well stick with p5 for the dom stuff but I am flexible about this if anyone thinks otherwise!

Reply:
I think if you're using p5 then yeah, use p5!